core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -384,15 +385,30 @@ private[spark] class MemoryStore(
     }
   }
 
+  private def maybeReleaseResources(resource: (BlockId, MemoryEntry[_])): Unit = {
+    maybeReleaseResources(resource._1, resource._2)
+  }
+
+  private def maybeReleaseResources(blockId: BlockId, entry: MemoryEntry[_]): Unit = {
+    entry match {
+      case SerializedMemoryEntry(buffer, _, _) => buffer.dispose()
```

Comment:
Why not just make these case classes `Closeable` and then you can close them consistently